### PR TITLE
Diagnose non-throwing methods implementing a throwing @objc protocol method

### PR DESCRIPTION
**Pull request by artemcm**
Resolves rdar://problem/60933487

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
Ah, that seems to be a ci failure. Will re-test. 
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -2456,6 +2456,9 @@ ERROR(override_throws,none,
 ERROR(override_throws_objc,none,
       "overriding a throwing @objc %select{method|initializer}0 with "
       "a non-throwing %select{method|initializer}0 is not supported", (bool))
+ERROR(implement_throws_objc,none,
+      "implementing a throwing @objc %select{method|initializer}0 with "
```
Comment (slavapestov) [inline include/swift/AST/DiagnosticsSema.def:2460]: maybe "satisfying a throwing @objc method requirement" is better phrasing here

**Review by slavapestov — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Pull request merged**